### PR TITLE
[ISSUE #4990]Fix test case: PlainPermissionManagerTest#multiFilePathTest

### LINKED ISSUE #4990: Fix test case: PlainPermissionManagerTest#multiFilePathTest
https://github.com/apache/rocketmq/runs/8173679384?check_suite_focus=true
image

### PR DESCRIPTION
**Pull request by mxsm**
**Make sure set the target branch to `develop`**

## What is the purpose of the change

Fix #4990 

## Brief changelog

- Fix test case: PlainPermissionManagerTest#multiFilePathTest

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Comment by Oliverwqcwrw** [comment]
Hello, @RongtongJin ,
I noticed that this PR has the same modification as the previous PR,
This pr #4637  adds a few test methods and changes  the file delimiter, PTAL

File: acl/src/test/java/org/apache/rocketmq/acl/plain/PlainPermissionManagerTest.java [hunk 1]
```diff
@@ -32,7 +32,9 @@
 import org.assertj.core.util.Lists;
 import org.junit.Assert;
 import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
```
Comment (RongtongJin) [inline acl/src/test/java/org/apache/rocketmq/acl/plain/PlainPermissionManagerTest.java:37]: This seems to be the imports of another pull request.

Comment (Oliverwqcwrw) [reply acl/src/test/java/org/apache/rocketmq/acl/plain/PlainPermissionManagerTest.java:37]: Should we think about merging #4637 and then #4991

File: acl/src/test/java/org/apache/rocketmq/acl/plain/PlainPermissionManagerTest.java [hunk 2]
```diff
@@ -215,16 +217,16 @@ public void isWatchStartTest() {
...
 
         PlainPermissionManager plainPermissionManager = new PlainPermissionManager();
 
-        String samefilePath = file.getAbsolutePath()+"/conf/acl/.";
-        String samefilePath2 = "/" +file.getAbsolutePath()+"/conf/acl";
-        String samefilePath3 = file.getAbsolutePath()+"/conf/acl/../"+file.getAbsolutePath();
-        String samefilePath4 = file.getAbsolutePath()+"/conf/acl///";
-        String samefilePath5 = file.getAbsolutePath()+"/conf/acl/./";
+        String samefilePath = file.getAbsolutePath()+"/conf/acl/.".replace("/", File.separator);
+        String samefilePath2 = file.getAbsolutePath()+"/conf/acl".replace("/", File.separator);
```
Comment (caigy) [inline acl/src/test/java/org/apache/rocketmq/acl/plain/PlainPermissionManagerTest.java:226]: The purpose of `samefilePath2` is to assure '/folder' is equivalent to '//folder' as path. IMO it should not be removed, but just check it in Linux or OS with '/' as path separator.

**Review by caigy — CHANGES_REQUESTED** [review]
It seems '/' is still a valid path separator in Windows, is it necessary to change it? Furthermore,  constructing paths by `File.separator` would be more preferred to replacing them one by one.

**Comment by Oliverwqcwrw** [comment]
Hello @mxsm ,
#4637 merged, need to resolve some conflicts

File: acl/src/test/java/org/apache/rocketmq/acl/plain/PlainPermissionManagerTest.java (hunk 2 above)
Comment (mxsm) [reply acl/src/test/java/org/apache/rocketmq/acl/plain/PlainPermissionManagerTest.java:236]: OK I will handle this case

File: acl/src/test/java/org/apache/rocketmq/acl/plain/PlainPermissionManagerTest.java [hunk 3]
```diff
@@ -53,10 +54,12 @@ public class PlainPermissionManagerTest {
     PlainAccessConfig plainAccessConfig = new PlainAccessConfig();
     Set<Integer> adminCode = new HashSet<>();
 
-    private static final String PATH = PlainPermissionManagerTest.class.getResource(File.separator).getFile();
+    private static final String PATH = "src/test/resources";
```
Comment (Oliverwqcwrw) [inline acl/src/test/java/org/apache/rocketmq/acl/plain/PlainPermissionManagerTest.java:57]: This does not seem to need to be modified to be compatible with multiple system separators

**Review by Oliverwqcwrw — CHANGES_REQUESTED** [review]

File: acl/src/test/java/org/apache/rocketmq/acl/plain/PlainPermissionManagerTest.java (hunk 3 above)
Comment (mxsm) [reply acl/src/test/java/org/apache/rocketmq/acl/plain/PlainPermissionManagerTest.java:57]: But when I run test case with befor code, not passed on windows platform

Comment (Oliverwqcwrw) [reply acl/src/test/java/org/apache/rocketmq/acl/plain/PlainPermissionManagerTest.java:57]: Regrettably, I don't have Windows system so I can't provide more information. I would like to know if this format `src/test/resources` supports Windows now

**Review by lizhanhui — CHANGES_REQUESTED** [review]
This pull request would regress to Linux only, breaking portability.

**Comment by mxsm** [comment]
> This pull request would regress to Linux only, breaking portability.

Got it
